pkg/ccl/streamingccl/streamproducer/producer_job.go

```diff
@@ -207,8 +207,7 @@ func updateReplicationStreamProgress(
 		})
 
 	if jobs.HasJobNotFoundError(err) || testutils.IsError(err, "not found in system.jobs table") {
-		status.StreamStatus = jobspb.StreamReplicationStatus_STREAM_INACTIVE
-		err = nil
+		err = errors.Wrapf(err, "Replication stream %d not found", streamID)
```

Comment:
Why this change?  I thought we wanted to communicate this information to the destination cluster not as an error but as an error status so that it can handle this appropriately?